pinboard.net.UnitTests/YesNoConverterSpec.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Moq;
+using Newtonsoft.Json;
+using pinboard.net.Util.Converters;
+using Xunit;
+
+namespace pinboard.net.UnitTests
+{
+    public class YesNoConverterSpec
+    {
+        [Theory]
+        [InlineData(typeof(string), true)]
+        [InlineData(typeof(int), false)]
+        [InlineData(typeof(decimal), false)]
+        [InlineData(typeof(DateTime), false)]
+        public void ShouldConvertOnlyCorrectType(Type type, bool expected)
+        {
+            var sut = new YesNoConverter();
+            var canConvert = sut.CanConvert(type);
+            Assert.Equal(expected, canConvert);
+        }
+
+        [Theory]
+        [InlineData(null, "no")]
+        [InlineData(false, "no")]
+        [InlineData(true, "yes")]
+        public void ShouldWriteCorrectValue(object value, string expected)
+        {
+            var jsonWriterMock = new Mock<JsonWriter>();
+            jsonWriterMock.Setup(m => m.WriteValue(It.IsAny<string>()));
+            var sut = new YesNoConverter();
+
+            sut.WriteJson(jsonWriterMock.Object, value, null);
+
+            jsonWriterMock.Verify(m => m.WriteValue(expected));
```

Comment:
This test here doesn't have an `Assert` call. So a failure of this test will be a failure of the `Verify` call. Am I right? 

Unsure if that is the right behaviour?

Reply:
Yes, you are right. Basically it's a Moq method to verify if the method was called with expected values. 

For example look at this [WriteJson](https://github.com/shrayasr/pinboard.net/blob/master/pinboard.net/Util/Converters/YesNoConverter.cs) method - it doesn't return any values nor change state so the only way to check if it is behaving as designed is to verify if it was called with expected values and that's exactly what the Verify method is doing.

Reply:
Got it 👍